src/components/NameListContext.js

```diff
@@ -1,17 +1,108 @@
 import React from "react";
+import axios from "axios";
 
-const { Provider, Consumer } = React.Context();
+let NameListContext;
+const { Provider, Consumer } = (NameListContext = React.createContext());
 
-class NameList extends React.Component {
-  state = {
-    list: [],
-    listView: [],
-    preferences: [],
+class NameListProvider extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      list: [
+        {
+          name: "",
+          year: 1880,
+          id: 14,
+          sex: "female",
+          popularity: 20,
+        },
+      ],
+      listView: [
+        {
+          name: "Helen",
+          year: 1998,
+          id: 14,
+          sex: "female",
+          popularity: 20,
+        },
+      ],
+      preferences: [],
+      sortBy: "name",
+      namesPerPage: 25,
+
+      onChangeOrder: this.onChangeOrder,
+      onChangeNumber: this.onChangeNumber,
+      onAdd: this.onAdd,
+      onDelete: this.onDelete,
+      callList: this.callList,
+    };
+  }
+
+  componentDidUpdate() {}
+
+  // Call the List from the Server => Sets List Component
+  callList = (listURL) => {
+    const callNames = async () => {
+      const names = await axios(listURL);
+      this.setState({ list: names.data, listView: names.data });
+      this.onChangeNumber(this.state.namesPerPage);
+    };
+    callNames();
+  };
+
+  // Ordering Names Switch Function
+  onChangeOrder = async (value) => {
+    if (value == "name") {
+      const results = [...this.state.list].sort((a, b) => {
+        let varA = a.name.toUpperCase();
```

Comment:
Use meaningful variable names when you can. Possibly `firstName` and `lastName` here.

Reply:
I agree, but this is only for first names...